subprojects/xcfa/xcfa-analysis/src/main/java/hu/bme/mit/theta/xcfa/analysis/impl/interleavings/XcfaPorLts.java

```diff
@@ -0,0 +1,135 @@
+package hu.bme.mit.theta.xcfa.analysis.impl.interleavings;
+
+import hu.bme.mit.theta.analysis.algorithm.PorLts;
+import hu.bme.mit.theta.core.decl.Decl;
+import hu.bme.mit.theta.core.decl.VarDecl;
+import hu.bme.mit.theta.core.type.Type;
+import hu.bme.mit.theta.xcfa.model.XCFA;
+import hu.bme.mit.theta.xcfa.model.XcfaEdge;
+import hu.bme.mit.theta.xcfa.model.XcfaLabel;
+import hu.bme.mit.theta.xcfa.model.XcfaLocation;
+import hu.bme.mit.theta.xcfa.model.utils.LabelUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class XcfaPorLts extends PorLts<XcfaState<?>, XcfaAction, XcfaEdge> {
+
+	private final XCFA xcfa;
+
+	private final XcfaLts simpleXcfaLts;
+
+	private final Random random = new Random();
```

Comment:
If not too complicated, the way to choose a transition should be configurable -- e.g. randomly (as it is now, if I understand correctly), round-robin style or one-by-one (eagerly explore a thread, then continue with another).